app/src/main/java/org/squidfish/tuplenback/models/GameViewModel.kt

```diff
@@ -115,18 +135,46 @@ class GameViewModel(private val repository: RecentRepository<GameModel>) : ViewM
     /**
      * Starts a new game. Also handles all necessary initialization
      */
-    private fun startGame(game: Game): Result<Unit, Error> {
-        this.game = game
+    private fun startGame(level: LevelData) {
+        viewModelScope.launch {
+            _gameLoadingState.value = GameLoadingState.Loading
 
-        gameEngines.clear()
+            val settings = when (val res = levelRepository.get(level)) {
+                is Result.Error -> {
+                    _gameLoadingState.value = GameLoadingState.Failiure(res.error)
+                    return@launch
+                }
 
-        initGameEngines().onError { return Result.Error(it) }
-        initGameState().onError { return Result.Error(it) }
+                is Result.Success -> res.data
+            }
 
-        Log.i(TAG, "Starting game")
-        startNewRound().onError { return Result.Error(it) }
+            val level = Level(
+                level = level.level,
+                game = level.gameMode,
+                settings = settings,
+            )
 
-        return Result.Success(Unit)
+            _level.value = level
+
+            gameEngines.clear()
+
+            initGameEngines().onError {
+                _gameLoadingState.value = GameLoadingState.Failiure(it)
+                return@launch
+            }
+            initGameState().onError {
+                _gameLoadingState.value = GameLoadingState.Failiure(it)
+                return@launch
```

Comment:
No need for the labels on the `return`s, there is nothing else in the function to run

Reply:
That doesn't seem to work. Apparently, you cannot have non-qualified return statements within lambda functions (which `viewModelScope.launch` takes), since there's compiler ambiguity from whence to return.